### PR TITLE
Add support for before/after time expression

### PR DESCRIPTION
**Pull request by EgoLaparra**
This pull request adds support to recognize time expressions of unbounded date ranges. 
It also covers properly time expressions with seasons a not a year (e.g. "since summer").

### CODE REVIEW

File: main/src/main/scala/org/clulab/numeric/mentions/package.scala [hunk 1]
```diff
@@ -369,11 +383,12 @@ package object mentions {
       if(seasonNorm.isEmpty)
         throw new RuntimeException(s"ERROR: could not find argument season in mention ${m.raw.mkString(" ")}!")
```
Comment (kwalcock) [inline main/src/main/scala/org/clulab/numeric/mentions/package.scala:384]: It doesn't especially matter in an error string, but I wonder why m.text is not used.  Perhaps it didn't exist at some point.

File: main/src/main/scala/org/clulab/numeric/mentions/package.scala [hunk 2]
```diff
@@ -339,11 +352,12 @@ package object mentions {
...
 
-      val yearNorm = getArgWords("year", m)
-      if(yearNorm.isEmpty)
-        throw new RuntimeException(s"ERROR: could not find argument year in mention ${m.raw.mkString(" ")}!")
-
-      val (_, yearEnd) = seasonNormalizer.adjustYearRange(seasonNorm.get, yearNorm.get)
+      val yearNorm = getArgWords("year", m) match {
+        case year if year.isDefined =>
```
Comment (kwalcock) [inline main/src/main/scala/org/clulab/numeric/mentions/package.scala:356]: Is this `case Some(year) =>` as above?

File: main/src/main/scala/org/clulab/numeric/mentions/DateRangeMention.scala [hunk 3]
```diff
@@ -19,8 +19,8 @@ class DateRangeMention ( labels: Seq[String],
   // does not have a year value but date2Norm has a year value
   // This will handle date ranges of form month date and month date of year.
   // Ex: "Sowing dates between May 3 and October 2, 2020" will have a normalized value of 2020-05-03 -- 2020-10-02
-  if (!date1Norm.equals("XXXX-XX-XX") && date1Norm.contains("XXXX")) {
-    if (!(date2Norm contains "XXXX")) {
+  if (!date1Norm.equals("XXXX-XX-XX") && date1Norm.startsWith("XXXX")) {
+    if (!(date2Norm.equals("ref-date") || date2Norm.startsWith("XXXX"))) {
```
Comment (kwalcock) [inline main/src/main/scala/org/clulab/numeric/mentions/DateRangeMention.scala:23]: Has someone been writing too much Java ;-) ?  You almost fooled me.  Is this `date1Norm != "XXXX-XX-XX"`?

**Review by kwalcock — COMMENTED** [review]
It looks really, really good.

File: main/src/main/scala/org/clulab/numeric/mentions/package.scala (hunk 1 above)
Comment (kwalcock) [reply main/src/main/scala/org/clulab/numeric/mentions/package.scala:384]: ...or maybe being raw was very important.

Comment (EgoLaparra) [reply main/src/main/scala/org/clulab/numeric/mentions/package.scala:384]: To be honest, I did not think much about it. I just followed the same code as the functions above.

File: main/src/main/scala/org/clulab/numeric/mentions/package.scala (hunk 2 above)
Comment (EgoLaparra) [reply main/src/main/scala/org/clulab/numeric/mentions/package.scala:356]: oops! yes, it should be the same.

File: main/src/main/scala/org/clulab/numeric/mentions/DateRangeMention.scala (hunk 3 above)
Comment (EgoLaparra) [reply main/src/main/scala/org/clulab/numeric/mentions/DateRangeMention.scala:23]: 😅 yes, you a right. I'll fix it right now!
By the way, looking at these 2 lines, the DateNorm companion object you were proposing makes even more sense.

**Comment by MihaiSurdeanu** [comment]
LGTM! @kwalcock: please merge when ready.

**Comment by kwalcock** [comment]
Thanks, @EgoLaparra.  I hope that wasn't too onerous.  You do good stuff.

Tests pass, so I'm pressing the button.

**Comment by MihaiSurdeanu** [comment]
Thanks @kwalcock and @EgoLaparra !

@mithunpaul08 : publishing processors locally should give you access to before/after date ranges.

**Comment by EgoLaparra** [comment]
@kwalcock good stuff? Thank goodness you keep an eye on every mess I try to push 😅 

@mithunpaul08 let me know if you run into any problem.

**Comment by mithunpaul08** [comment]
I agree with @egolaparra. @kwalcock is like the universal try catch block of the lab. If it was not for him i can't imagine how many of my accidental mistakes would have been in production already.

**Pull request merged**